tests/unit/alp_cholesky.cpp

```diff
@@ -22,71 +22,169 @@
+alp::RC check_cholesky_solution(
+	const  alp::Matrix< T, structures::Symmetric, Dense > &H,
+	alp::Matrix< T, structures::UpperTriangular, Dense > &L,
+	const ZeroType &zero_scalar,
+	const RingType &ring
+) {
+	alp::RC rc = SUCCESS;
+	const size_t N = nrows( H );
+	alp::Matrix< T, alp::structures::Symmetric > LLT( N, N );
+	alp::set( LLT, zero_scalar );
+	auto LT = alp::get_view< alp::view::transpose >( L );
+#ifdef DEBUG
+	print_matrix( " << LLT >> ", LLT );
+	print_matrix( " << LT >>  ", LT );
+#endif
+	alp::mxm( LLT, LT, L, ring );
+#ifdef DEBUG
+	print_matrix( " << LLT >> ", LLT );
+#endif
+
+	alp::Matrix< T, alp::structures::Symmetric > TMP( N, N );
+	alp::set( TMP, zero_scalar );
+
+	// LLT = -LLT
+	alp::Scalar< T > alpha( -1 );
+	rc = rc ? rc : alp::eWiseLambda(
+		[ ]( const size_t i, const size_t j, T &val ) {
+			(void)i;
+			(void)j;
+			val = -val;
+		},
+		LLT
+	);
+#ifdef DEBUG
+	print_matrix( " << -LLT  >> ", LLT );
+#endif
+
+	// TMP = H - LLT
+	rc = rc ? rc : alp::eWiseApply(
+		TMP, H, LLT,
+		ring.getAdditiveMonoid()
+	);
+#ifdef DEBUG
+	print_matrix( " << H - LLT  >> ", TMP );
+#endif
+
+	//Frobenius norm
+	T fnorm = 0;
+	rc = rc ? rc : alp::eWiseLambda(
+		[ &fnorm ]( const size_t i, const size_t j, T &val ) {
+			(void)i;
+			(void)j;
+			fnorm += val*val;
+		},
+		TMP
+	);
+	fnorm = std::sqrt(fnorm);
+	std::cout << " FrobeniusNorm(H-LL^T) = " << fnorm << "\n";
+	if ( tol < fnorm ) {
+		throw std::runtime_error(
+			"The Frobenius norm is too large. "
+			"Make sure that you have used SPD matrix as input."
+		);
+	}
 
-	alp::utils::MatrixFileReader<
-		double
-	> parser_A( std::string("/home/d/Repos/graphblas/datasets/mymatrix.mtx") );
+	return rc;
+}
 
-	for ( auto it = parser_A.begin() ; it != parser_A.end() ; ++it  ) {
-		std::cout << " i,j,v= " << it.i() << " " << it.j() << " " << it.v() << "\n";
-	}
+void alp_program( const inpdata &unit, alp::RC &rc ) {
+	rc = SUCCESS;
+	auto fname = unit.fname;
+	alp::utils::MatrixFileReader< ScalarType > parser_A( fname );
+	size_t N = parser_A.n();
+
+	alp::Semiring<
+		alp::operators::add< ScalarType >,
+		alp::operators::mul< ScalarType >,
+		alp::identities::zero,
+		alp::identities::one
+	> ring;
+	alp::Scalar< ScalarType > zero_scalar( ring.getZero< ScalarType >() );
 
-	return ;
+	std::cout << "\tTesting ALP cholesky\n"
+		"\tH = L * L^T\n";
 
+	alp::Matrix< ScalarType, structures::Symmetric, Dense > H( N, N );
+	alp::Matrix< ScalarType, structures::UpperTriangular, Dense > L( N, N );
 
+	if( !parser_A.isSymmetric() ) {
+		std::cout << "Symmetric matrix epxected as input!\n";
+		rc = ILLEGAL;
+		return;
+	}
 
-	alp::Semiring< alp::operators::add< double >, alp::operators::mul< double >, alp::identities::zero, alp::identities::one > ring;
+	rc = rc ? rc : alp::buildMatrix( H, parser_A.begin(), parser_A.end() );
 
-	std::cout << "\tTesting ALP cholesky\n"
-	             "\tH = L * L^T\n";
+	if( !internal::getInitialized( H ) ) {
```

Comment:
Is this a debug check? If buildMatrix was successful, `H` should be initialized. If `H` is not initialized, should the function return at this point?